src/msg/async/PosixStack.cc

```diff
@@ -119,13 +119,12 @@ class PosixConnectedSocketImpl final : public ConnectedSocketImpl {
       uint64_t size = MIN(left_pbrs, IOV_MAX);
       left_pbrs -= size;
       memset(&msg, 0, sizeof(msg));
-      msg.msg_iovlen = 0;
+      msg.msg_iovlen = size;
       msg.msg_iov = msgvec;
       unsigned msglen = 0;
       while (size > 0) {
         msgvec[msg.msg_iovlen].iov_base = (void*)(pb->c_str());
```

Comment:
we are using `msg.msg_iovlen` to index `msgvec`, probably we could put something like:
```c++
for (auto iov = msgvec; iov != msgvec + size; iov++) {
  iov->iov_base = (void*)(pb->c_str());
  iov->iov_len = pb->length();
  msglen += pb->length();
  ++pb;
}
```